staging/src/k8s.io/legacy-cloud-providers/azure/clients/vmssclient/azure_vmssclient.go

```diff
@@ -272,6 +272,38 @@ func (c *Client) CreateOrUpdateAsync(ctx context.Context, resourceGroupName stri
 	return future, nil
 }
 
+// WaitForCreateOrUpdateResult waits for the response of the create or update request
+func (c *Client) WaitForCreateOrUpdateResult(ctx context.Context, future *azure.Future, resourceGroupName string) (*http.Response, error) {
```

Comment:
this and the delete one were added to better track metrics. I'll remove `WaitForAsyncResult` once autoscaler switches to using `WaitForCreateOrUpdateResult` and `WaitForDeleteInstancesResult`